mws/utils/collections.py

```diff
@@ -0,0 +1,109 @@
+"""Data structure utilities."""
+
+from collections.abc import Mapping, Iterable
+
+
+def unique_list_order_preserved(seq):
+    """Returns a unique list of items from the sequence
+    while preserving original ordering.
+    The first occurrence of an item is returned in the new sequence:
+    any subsequent occurrences of the same item are ignored.
+    """
+    seen = set()
+    seen_add = seen.add
```

Comment:
why the seen_add here? Whats wrong with just adding to the set?

Reply:
I had forgotten why at the time, but the answer is a performance boost: https://stackoverflow.com/a/46422008

> Using a local reference is almost 3 times as fast. Because the set.add is used in a loop, it is worth optimising away the attribute lookup.

Reply:
Thats interesting!